doc/perception_pipeline/perception_pipeline_tutorial.rst

```diff
@@ -106,3 +122,63 @@ MoveIt! uses an octree-based framework to represent the world around it. The *Oc
  * *octomap_frame*: specifies the coordinate frame in which this representation will be stored. If you are working with a mobile robot, this frame should be a fixed frame in the world.
  * *octomap_resolution*: specifies the resolution at which this representation is maintained (in meters).
  * *max_range*: specifies the maximum range value to be applied for any sensor input to this node.
+
+Obstacle Avoidance
+------------------
+
+If you set the initial and the final location of the robot in a way that there is no straight path between them, then the planner will automatically avoid the octomap and plan around it.
+
+.. image:: obstacle_avoidance.gif
+   :width: 700px
+
+Running the Interface
++++++++++++++++++++++
+Roslaunch the launch file to run the code directly from moveit_tutorials: ::
+
+ roslaunch moveit_tutorials obstacle_avoidance_demo.launch
+
+you should see something like the image shown at the beginning of this tutorial.
+
+You can test obstacle avoidance for yourself by setting the goal state manually and then planning and executing. To learn how to do that look at `MoveIt! Quickstart in RViz <../quickstart_in_rviz/quickstart_in_rviz_tutorial.html>`_
+
+Detecting and Adding Object as Collision Object
+-----------------------------------------------
+
+In this section, we will demonstrate an example of extracting a cylinder from a pointcloud, computing relevant values and adding it as a collision object to the planning scene. 
+We will be working with point clouds but it can be implemented similarly with depth maps.
+
+After running the code, you should be able to see something like this in rviz:
+
+.. image:: cylinder_collision_object.png
+   :width: 700px
+
+Running the Code
+++++++++++++++++
+Roslaunch the launch file to run the code directly from moveit_tutorials: ::
+
+ roslaunch moveit_tutorials detect_and_add_cylinder_collision_object_demo.launch
+
+KNOWN ISSUE - You may see the following error when running the demo ::
+
+  ros.moveit_ros_planning.planning_scene_monitor: Transform error: Lookup would require extrapolation into the future.  Requested time 1527473962.793050157 but the latest data is at time 1527473962.776993978, when looking up transform from frame [panda_link2] to frame [camera_rgb_optical_frame]
+  ros.moveit_ros_perception: Transform cache was not updated. Self-filtering may fail.
+
+We are working on fixing it, it should not break the working of the demo.
```

Comment:
Is there a Github issue tracking the progress of this problem? Please link to it (and create it if needed)

Reply:
[created](https://github.com/ros-planning/moveit_tutorials/issues/192)